core/src/main/java/hudson/model/Slave.java

```diff
@@ -373,7 +375,7 @@ private URLConnection connect() throws IOException {
             return res.openConnection();
         }
 
-        public URL getURL() throws MalformedURLException {
+        public URL getURL() throws IOException {
```

Comment:
Actually it is binary compatible, if not source compatible. Really this method is only `public` for tests and should have been restricted.